example/lib/main.dart

```diff
@@ -60,11 +60,9 @@ class _MyAppState extends State<MyApp> {
 
   // Platform messages are asynchronous, so we initialize in an async method.
   Future<void> initPlatformState() async {
-    await Future.wait([
-      Permission.location.request(),
-      Permission.locationAlways.request(),
-      Permission.notification.request(),
-    ]);
+    await Permission.location.request();
+    await Permission.locationAlways.request();
+    await Permission.notification.request();
```

Comment:
Having a Future that awaits on a list of async calls performs better than having to await for each and every async call

in the old implementation if the first call took 1 ms and the second 2 and the third 3, then we will wait for only 3 seconds, but in this implementation we will wait for 6 seconds

Reply:
If more than one of these permissions are not yet granted, calling these async functions in parallel results in the following error:

```
E/flutter ( 9876): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(PermissionHandler.PermissionManager, A request for permissions is already running, please wait for it to finish before doing another request (note that you can request multiple permissions at the same time)., null, null)
E/flutter ( 9876): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:648:7)
E/flutter ( 9876): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:334:18)
E/flutter ( 9876): <asynchronous suspension>
E/flutter ( 9876): #2      MethodChannelPermissionHandler.requestPermissions (package:permission_handler_platform_interface/src/method_channel/method_channel_permission_handler.dart:80:9)
E/flutter ( 9876): <asynchronous suspension>
E/flutter ( 9876): #3      PermissionActions.request (package:permission_handler/permission_handler.dart:109:10)
E/flutter ( 9876): <asynchronous suspension>
E/flutter ( 9876): #4      Future.wait.<anonymous closure> (dart:async/future.dart:524:21)
E/flutter ( 9876): <asynchronous suspension>
E/flutter ( 9876): #5      _MyAppState.initPlatformState (package:geofence_foreground_service_example/main.dart:67:5)
E/flutter ( 9876): <asynchronous suspension>
E/flutter ( 9876): 
```